### ISSUE TITLE
(3P) Podcast Player: fix React memo optimizations

### ISSUE THREAD

**Issue by marekhrabe** [Automattic/jetpack#15272]
Most of our components have very simple and deterministic output so I've wrapped them in `memo`. Introducing the `colors` object that regenerates them with every render has removed all advantages of `memo` and needs to re-render every single time as props are changing.

Let's optimize how components handle the properties and their propagation.

Feedback:

> I was thinking whether it would make sense to convert all those const declarations you create from scratch in each render into an effect hook.
> - [Marek](https://github.com/Automattic/jetpack/pull/15249#issuecomment-607761226)

> Perhaps we should consider creating a colors Context and then using that throughout the Block so that we can access without prop drilling? Indeed this is almost exactly the scenario the React team provide in their Context docs.
> [Dave](https://github.com/Automattic/jetpack/pull/15249#pullrequestreview-386321944)

**Comment by obenland** [comment]
Let's start with an effect hook and go from there?

**Comment by marekhrabe** [comment]
When I suggested an effect hook, I haven't realized we are in a class component. I believe class component makes sense for relatively complex component like this. 

So hooks were out the window. The second logical option is `getDerivedStateFromProps` but what I would implement is checking if any color prop has changed and update state in such case. Which is 100% what [React team marked as an anti-pattern](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization) and recommends using memoization instead. So that's what I did :)
